src/Refactoring-Core/RBRemoveClassRefactoring.class.st

```diff
@@ -75,13 +96,28 @@ RBRemoveClassRefactoring >> classNames: aClassNameCollection [
 	classNames := aClassNameCollection
 ]
 
+{ #category : 'preconditions' }
+RBRemoveClassRefactoring >> eagerlyCheckApplicabilityPreconditions [
+
+	self applicabilityPreconditions
+		do: [ :cond | cond check ifFalse:
+				[ self refactoringError:
+					(String streamContents: [ :aStream | cond violationMessageOn: aStream ] ) ] ]
+]
+
 { #category : 'preconditions' }
 RBRemoveClassRefactoring >> environmentWithUsersOf: aClassable [
 	^ RBClassEnvironment
 		onEnvironment: RBBrowserEnvironment new
 		classes: (self model classesReferencingClass: aClassable)
 ]
 
+{ #category : 'preconditions' }
+RBRemoveClassRefactoring >> preconditionAreNotMeta [
+
+	^ RBClassesAreNotMetaClassCondition new classes: classesDictionary
```

Comment:
It could be nice to add #classes: on the class side. And if this is done there is no need to have this method I think. `RBClassesAreNotMetaCondition classes: classesDictionary` is readable enought

Reply:
Agreed, I had it already in my Todo but forgot it :(